File: Source Code/Samples/Bölüm 7/Rethrow/sample.cs
namespace DotNetKitabý.Exceptions
{
    using System.Globalization;
    using System.Threading;
    using System.Text;
    using System.Collections.Generic;
    using System;
    using System.IO;
    using System.Collections;
    using System.Web;


    static class Program
    {
        

        static void Test()
        {
           throw new InvalidDataException("X");
        }

        static void Test1()
        {
            try
            {
                Test();
            }
            catch 
            {
                throw;
            }
        }

        static void Test2()
        {
            try
            {
                Test();
            }
            catch (Exception exc)
            {
                
                throw;
            }
        }

        static void Test3()
        {
            try
            {
                Test();
            }
            catch (Exception exc)
            {

                throw exc;
            }
        }


        static void Main(string[] args)
        {
            try
            {
                Test2();
            }
            catch (Exception exc)
            {
                Console.WriteLine(exc.StackTrace);
            }

            Console.WriteLine("---------------------------------------------");

            try
            {
                Test3();
            }
            catch (Exception exc)
            {
                Console.WriteLine(exc.StackTrace);
            }

            Console.ReadLine();
        }

    }
}

File: Source Code/Samples/Bölüm 7/Rethrow/sample.vb
Imports System.Globalization
Imports System.Threading
Imports System.Text
Imports System.Collections
Imports System
Imports System.IO


Class Program

    Shared Sub Test()
        Throw New InvalidDataException("X")
    End Sub

    Shared Sub Test1()
        Try
            Test()
        Catch
            Throw
        End Try
    End Sub

    Shared Sub Test2()
        Try
            Test()
        Catch exc As Exception
            Throw
        End Try
    End Sub

    Shared Sub Test3()
        Try
            Test()
        Catch exc As Exception
            Throw exc
        End Try
    End Sub

    Shared Sub Main(ByVal args() As String)
        Try
            Test2()
        Catch exc As Exception
            Console.WriteLine(exc.StackTrace)
        End Try

        Console.WriteLine("-------------------------------------")

        Try
            Test3()
        Catch exc As Exception
            Console.WriteLine(exc.StackTrace)
        End Try
        Console.ReadLine()
    End Sub


End Class

